source/virtualBuffers/gecko_ia2.py

```diff
@@ -18,10 +18,58 @@
 from comtypes import COMError
 import aria
 import config
-from NVDAObjects.IAccessible import normalizeIA2TextFormatField
+from NVDAObjects.IAccessible import normalizeIA2TextFormatField, IA2TextTextInfo, getNVDAObjectFromPoint
 
 class Gecko_ia2_TextInfo(VirtualBufferTextInfo):
 
+	def _getPointFromOffset(self,offset):
+		formatFieldStart, formatFieldEnd = self._getUnitOffsets(self.UNIT_FORMATFIELD, offset)
+		# The format field starts at the first character.
+		for field in reversed(self._getFieldsInRange(formatFieldStart, formatFieldStart+1)):
+			if not (isinstance(field, textInfos.FieldCommand) and field.command == "formatChange"):
+				# This is no format field.
+				continue
+			attrs = field.field
+			ia2TextStartOffset = attrs.get("ia2TextStartOffset")
+			if ia2TextStartOffset is None:
+				continue
+			ia2TextStartOffset += attrs.get("strippedCharsFromStart", 0)
+			relOffset = offset - formatFieldStart + ia2TextStartOffset
+			obj = self._getNVDAObjectFromOffset(offset)
+			if not issubclass(obj.TextInfo, IA2TextTextInfo):
+				raise LookupError("TextInfo is of type %r and does not provide reliable points" % obj.TextInfo)
+			# Position is irrelevant here, we just need a text info to work with
+			info = obj.makeTextInfo(textInfos.POSITION_FIRST)
+			return info._getPointFromOffset(relOffset)

```

Comment:
I'm not a fan of having to instantiate this here, especially given the above. Perhaps you could make this a classmethod or split it into a helper function. The former should work, but would need testing. (You should be able to get away with only making it a classmethod for that one TextInfo, rather than the whole of NVDA.)